### PR TITLE
MacOS .dmg fixes

### PR DESCRIPTION
**Pull request by Meakk**
Fixes both issues of https://github.com/f3d-app/f3d-superbuild/issues/173

Also fix an issue when `CMAKE_OSX_ARCHITECTURE` is not set.

### CODE REVIEW

File: projects/apple/f3d.bundle.cmake [hunk 1]
```diff
@@ -56,5 +56,14 @@ install(
 
 ## DragNDrop Generator specific
 if (cpack_generator MATCHES "DragNDrop")
+
+  # Configure to replace the app name in the script.
+  configure_file(
+    "${CMAKE_CURRENT_LIST_DIR}/files/f3d-dmg-setup.scpt.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/f3d-dmg-setup.scpt"
```
[the remark this answers is not in the document]
Comment (Meakk) [reply projects/apple/f3d.bundle.cmake:63]: There is a @f3d_name@

[the remark this answers is not in the document]
Comment (mwestphal) [reply projects/apple/f3d.bundle.cmake:63]: indeed!

**Review by mwestphal — APPROVED** [review]

File: projects/f3d.bundle.common.cmake [hunk 2]
```diff
@@ -14,7 +14,7 @@ set(CPACK_PACKAGE_VENDOR "${f3d_name}-app")
 set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "${f3d_description}")
 set(CPACK_PACKAGE_HOMEPAGE_URL ${f3d_url})
 set(CPACK_PACKAGE_CHECKSUM MD5)
-set(CPACK_PACKAGE_EXECUTABLES f3d f3d)
+set(CPACK_PACKAGE_EXECUTABLES f3d)
```
Comment (mwestphal) [inline projects/f3d.bundle.common.cmake:17]: this is breaking the Windows CI

Comment (mwestphal) [reply projects/f3d.bundle.common.cmake:17]: ` CPack Error: CPACK_PACKAGE_EXECUTABLES should contain pairs of <executable> and <icon name>.`

**Review by mwestphal — CHANGES_REQUESTED** [review]
CI not clean

File: projects/f3d.bundle.common.cmake (hunk 2 above)
Comment (Meakk) [reply projects/f3d.bundle.common.cmake]: indeed

**Review by mwestphal — APPROVED** [review]

**Pull request merged**